src/Shell/Process.php

```diff
@@ -479,6 +562,26 @@ public function read()
         $this->outputHandler->handle($stdout, $stderr);
     }
 
+    /**
+     * @return array
+     */
+    public function getExpectedExitcodes()
+    {
+        return $this->expectedExitcodes;
+    }
+
+    /**
+     * @param array $expectedExitcodes
+     *
+     * @return $this
+     */
+    public function setExpectedExitcodes(array $expectedExitcodes)
+    {
+        $this->expectedExitcodes = $expectedExitcodes;
```

Comment:
Changing this to

```php
$this->expectedExitcodes = array_flip($expectedExitcodes)
````

will improve the exit code check - a hash table lookup is better than iterating an array.

```php
in_array($this->exitcode, $this->expectedExitcodes);
```

v

```php
array_key_exists($this->exitcode, $this->expectedExitcodes);
// or isset($this->expectedExitcodes[$this->exitcode]);
```

The definition of getExpectedExitcodes can be changed to

```php
return array_keys($this->expectedExitcodes);
```
in order to maintain the expected array value.

Reply:
you're right, `in_array` is not the fastest way, but imho `in_array` is more readable and the performance impact is not worth it